Java/src/com/couchbase/litecore/C4Replicator.java

```diff
@@ -61,19 +66,26 @@
                  int push, int pull,
                  byte[] options,
                  C4ReplicatorListener listener,
+                 C4ReplicationFilter pushFilter,
+                 C4ReplicationFilter pullFilter,
                  Object replicatorContext,
                  int socketFactoryContext,
                  int framing) throws LiteCoreException {
         this.listener = listener;
         this.context = replicatorContext; // replicator context
+        this.pushFilter = pushFilter;
+        this.pullFilter = pullFilter;
         handle = createV2(db, schema, host, port, path, remoteDatabaseName,
                 otherLocalDB,
                 push, pull,
                 replicatorContext.hashCode(),
                 framing,
                 socketFactoryContext,
+                pushFilter,
+                pullFilter,
                 options);
         reverseLookupTable.put(handle, this);
+        reverseLookupTable2.put((long)socketFactoryContext, this);
```

Comment:
IMO, the original code before this PR is very confusing. It seems like `replicatorContext.hashCode() == socketFactoryContext`. It's hard to follow from the code review where both replicatorContext and socketFactoryContext get used. We really need to clean this up.

As Java_com_couchbase_litecore_C4Replicator_createV2 sets the callback context to the replicatorContext.hashCode(), the correct value to set here is `replicatorContext.hashCode()`.